src/api/cvc4cpp.cpp

```diff
@@ -2073,6 +2073,109 @@ CVC4::Expr Term::getExpr(void) const
 // to the new API. !!!
 const CVC4::Node& Term::getNode(void) const { return *d_node; }
 
+namespace detail {
+const Rational& getRational(const CVC4::Node& node)
+{
+  return node.getConst<Rational>();
+}
+Integer getInteger(const CVC4::Node& node)
+{
+  return node.getConst<Rational>().getNumerator();
+}
+template <typename T>
+bool checkIntegerBounds(const Integer& i)
+{
+  return i >= std::numeric_limits<T>::min()
+         && i <= std::numeric_limits<T>::max();
+}
+bool checkReal32Bounds(const Rational& r)
+{
+  return checkIntegerBounds<std::int32_t>(r.getNumerator())
+         && checkIntegerBounds<std::uint32_t>(r.getDenominator());
+}
+bool checkReal64Bounds(const Rational& r)
+{
+  return checkIntegerBounds<std::int64_t>(r.getNumerator())
+         && checkIntegerBounds<std::uint64_t>(r.getDenominator());
+}
+
+bool isReal(const Node& node)
```

Comment:
So this may lead to some confusion.

There are actually two ways of representing real constants at the Term level.  A Term is a constant rational if its Node is CONST_RATIONAL{r} with a non-integer r, or (CAST_TO_REAL CONST_RATIONAL{r}) where r is an integral rational.  This is handled in appropriate places https://github.com/CVC4/CVC4/blob/master/src/api/cvc4cpp.h#L1178.
This is what allows us to hide subtyping.

I'd recommend for now inlining this method into isInteger.